Client/Frontend/Settings/Brave Today/BraveTodayAddSourceViewController.swift

```diff
@@ -0,0 +1,371 @@
+        }
+        present(picker, animated: true)
+    }
+    
+    private func rssLocationFromOPMLOutline(_ outline: OPML.Outline) -> RSSFeedLocation? {
+        guard let url = outline.xmlUrl?.asURL else { return nil }
+        return .init(title: outline.text, url: url)
+    }
+    
+    private let session: URLSession = {
+        let configuration = URLSessionConfiguration.default
+        configuration.timeoutIntervalForRequest = 5
+        return URLSession(configuration: configuration, delegate: nil, delegateQueue: .main)
+    }()
+    
+    private func displayError(_ error: FindFeedsError) {
+        let alert = UIAlertController(title: Strings.BraveToday.addSourceFailureTitle, message: error.localizedDescription, preferredStyle: .alert)
+        alert.addAction(.init(title: Strings.OKString, style: .default, handler: nil))
+        present(alert, animated: true)
+    }
+    
+    private func searchPageForFeeds() {
+        guard var text = textField.text else { return }
+        if text.hasPrefix("feed:"), let range = text.range(of: "feed:") {
+            text.replaceSubrange(range, with: [])
+        }
+        guard let url = URIFixup.getURL(text) else { return }
+        downloadPageData(for: url) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let data):
+                let resultsController = BraveTodayAddSourceResultsViewController(
+                    dataSource: self.feedDataSource,
+                    searchedURL: url,
+                    rssFeedLocations: data,
+                    sourcesAdded: self.sourcesAdded
+                )
+                self.navigationController?.pushViewController(resultsController, animated: true)
+            case .failure(let error):
+                self.displayError(error)
+            }
+        }
+    }
+    
+    private enum FindFeedsError: Error {
+        /// An error occured while attempting to download the page
+        case dataTaskError(Error)
+        /// The data was either not received or is in the incorrect format
+        case invalidData
+        /// The data downloaded did not match a
+        case parserError(ParserError)
+        /// No feeds were found at the given URL
+        case noFeedsFound
+        
+        var localizedDescription: String {
+            switch self {
+            case .dataTaskError(let error as URLError) where error.code == .notConnectedToInternet:
+                return error.localizedDescription
+            case .dataTaskError:
+                return Strings.BraveToday.addSourceNetworkFailureMessage
+            case .invalidData, .parserError:
+                return Strings.BraveToday.addSourceInvalidDataMessage
+            case .noFeedsFound:
+                return Strings.BraveToday.addSourceNoFeedsFoundMessage
+            }
+        }
+    }
+    
+    private var pageTask: URLSessionDataTask?
```

Comment:
Maybe worth cancelling this task on deinit

Reply:
For some reason I thought `URLSessionDataTask.deinit` calls `cancel` automatically… I'll test this